### PR TITLE
Update role-based-access-control.md

### PR DESCRIPTION
**Pull request by xyqcmss**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
用 `rw_user1@localhost` 登陆，tidb中无法通过localhost登陆，需要在tidb-server的机器上指定127.0.0.1登陆：
mysql -h 127.0.0.1 -P 4000 -u rw_user1 -prw_user1pass;

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [x] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: role-based-access-control.md [hunk 1]
```diff
@@ -278,10 +278,14 @@ SET ROLE ALL EXCEPT 'app_read'
 SET DEFAULT ROLE ALL TO 'rw_user1'@'localhost';
 ```
 
-用 `rw_user1@localhost` 登陆后：
+用 `rw_user1@localhost` 登陆，TiDB中无法通过 `localhost` 登陆，需要在 tidb-server 机器上指定127.0.0.1登陆：
```
Comment (TomShawn) [inline role-based-access-control.md:281]:
```suggestion
用 `rw_user1@localhost` 登陆。TiDB 中 无法通过 `localhost` 登陆，需要在 tidb-server 机器上指定 `127.0.0.1` 后才可登陆：
```

File: role-based-access-control.md [hunk 2]
```diff
@@ -278,10 +278,14 @@ SET ROLE ALL EXCEPT 'app_read'
...
 
 {{< copyable "sql" >}}
```
Comment (TomShawn) [inline role-based-access-control.md:283]:
```suggestion
{{< copyable "shell-regular" >}}
```

File: role-based-access-control.md [hunk 3]
```diff
@@ -278,10 +278,14 @@ SET ROLE ALL EXCEPT 'app_read'
...
 
+```sql
```
Comment (TomShawn) [inline role-based-access-control.md:285]:
```suggestion
```shell
```

**Comment by TomShawn** [comment]
@tiancaiamao PTAL and help determine all applicable versions. Thanks!

File: role-based-access-control.md (hunk 1 above)
Comment (tiancaiamao) [reply role-based-access-control.md:281]: > TiDB 中 无法通过 `localhost` 登陆

No, TiDB can use `localhost`, you need to configure the unix domain socket by setting [here](https://github.com/pingcap/tidb/blob/3ed4a23b956da63c95a5665432fef687a096601c/config/config.toml.example#L19)

```
host = ""
port = 0
socket = "/var/run/mysqld/mysqld.sock"
```

If you modify those configure items, you can do use localhost.


Or maybe you can update some of the related content to `127.0.0.1` if you find this guide confuse you.

**Comment by xyqcmss** [comment]
\cc @tiancaiamao 
Thank you for your guidance. Now there is a new problem that users of dev1 can log in on localhost, but users of dev3 cannot log in:
image

**Comment by tiancaiamao** [comment]
Try this configuration:

```
host = "0.0.0.0"
port = 4000
socket = "/var/run/mysqld/mysqld.sock"
```

And if you find something unnatural, you can just file a issue to the TiDB repo.
We will take a look to verify whether it's a bug.

@xyqcmss

**Comment by xyqcmss** [comment]
\cc @tiancaiamao
Thank you very much, I will continue to study

**Pull request closed without merging**